src/operator/contrib/yolo_output-inl.h

```diff
@@ -0,0 +1,703 @@
+     Tensor<xpu, 3, DType> temp_anchors(xpu_bias.dptr_ + xpu_bias.MSize(),
+      anchors_shape, s);
+     ScalarExp<DType> in_w = ScalarExp<DType>(1.0 / data.shape_[3]);
+     ScalarExp<DType> in_h = ScalarExp<DType>(1.0 / data.shape_[2]);
+
+     // change the order of dimensions;
+     temp = reshape(transpose(data, Shape4(0, 2, 3, 1)), temp.shape_);
+     softmax_in = transpose(slice<2>(temp, 0, nc), Shape3(0, 2, 1));
+     Softmax(softmax_out, softmax_in);
+     slice<2>(temp, 0, nc) = transpose(softmax_out, Shape3(0, 2, 1));
+     // class id to output
+     slice<2>(out, 0, 1) = transpose(reduce_keepdim<red::maximum, true>(
+       softmax_out, 1), Shape3(0, 2, 1));
+     // apply logistic to score, x, y
+     slice<2>(temp, nc, nc + 3) = F<mshadow_op::sigmoid>(slice<2>(temp, nc, nc + 3));
+     // scores to output
+     slice<2>(out, 1, 2) = F<mshadow_op::identity>(slice<2>(
+      temp, nc, nc + 1));
+     // x = (logistic(pred[0]) + i) / in_w
+     tshape[2] = 1;
+     slice<2>(out, 2, 3) = in_w * (slice<2>(temp, nc + 1, nc + 2) +
+      reshape(broadcast_with_axis(repmat(range<DType>(
+      0, data.shape_[3], 1, param_.num_anchor), data.shape_[2]), -1,
+      data.shape_[0]), tshape));
+     // y = (logistic(pred[1]) + j) / in_h
+     slice<2>(out, 3, 4) = in_h * (slice<2>(temp, nc + 2, nc + 3) +
+      reshape(broadcast_with_axis(range<DType>(0,
+      data.shape_[2], 1, data.shape_[3] * param_.num_anchor), -1, data.shape_[0]),
+      tshape));
+     // anchors
+     nnvm::Tuple<DType> anchors(param_.anchors.begin(), param_.anchors.end());
+     Tensor<cpu, 1, DType> cpu_bias(anchors.begin(), Shape1(anchors.ndim()));
+     Copy(xpu_bias, cpu_bias, s);
+     temp_anchors = reshape(repmat(xpu_bias, data.shape_[0] * data.shape_[2] * data.shape_[3]),
+      temp_anchors.shape_);
+     // w = exp(pred[2]) * anchor[w] / in_w
+     slice<2>(out, 4, 5) = in_w * F<mshadow_op::exp>(slice<2>(temp, nc + 3, nc + 4)) *
+      slice<2>(temp_anchors, 0, 1);
+     // h = exp(pred[3]) * anchor[y] / in_h
+     slice<2>(out, 5, 6) = in_h * F<mshadow_op::exp>(slice<2>(temp, nc + 4, nc + 5)) *
+      slice<2>(temp_anchors, 1, 2);
+
+     // convert output from x, y, w, h to xmin, ymin, xmax, ymax format
+     slice<2>(out, 2, 3) -= ScalarExp<DType>(0.5) * slice<2>(out, 4, 5);
+     slice<2>(out, 3, 4) -= ScalarExp<DType>(0.5) * slice<2>(out, 5, 6);
+     slice<2>(out, 4, 5) += slice<2>(out, 2, 3);
+     slice<2>(out, 5, 6) += slice<2>(out, 3, 4);
+
+     // make copy for backward
+     out_copy = F<mshadow_op::identity>(out);
+
+     // clip to boundaries
+     slice<2>(out, 2, 6) = F<clip_zero_one>(slice<2>(out, 2, 6));
+
+     // apply nms
+     if (param_.nms_threshold > 0 && param_.nms_threshold < 1) {
+       int keep = param_.nms_topk < 0 ? out.shape_[1] : param_.nms_topk;
+       keep = keep > static_cast<int>(out.shape_[1]) ? out.shape_[1] : keep;
+       if (keep > 0) {
+         // descend sort by score
+         int num_batch = out.shape_[0];
```

Comment:
same comment, why don't we leave those as the original index_t unsigned type?

Reply:
negative value is supported on purpose